src/Exception/RequestException.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Http\Client\Exception;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Thrown when a request is available
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+class RequestException extends TransferException
+{
+    /**
+     * @var RequestInterface
+     */
+    protected $request;
```

Comment:
why does this need to be protected? the request should be injected in the constructor and we have a getter method here. private should work.